crates/bitwarden-state/README.md

```diff
@@ -39,21 +39,20 @@ need to define some functions in `bitwarden-wasm-internal` and `bitwarden-uniffi
 applications to provide their `Repository` implementations. The implementations themselves will be
 very simple as we provide macros that take care of the brunt of the work.
 
-### Client-Managed State in WASM
+To add support for a new `Repository`, add it to the list defined at the end of
+`crates/bitwarden-pm/src/migrations.rs`.
 
-For WASM, we need to define a new `Repository` for our type and provide a function that will accept
-it. This is done in the file `crates/bitwarden-wasm-internal/src/platform/mod.rs`, you can check the
-provided example:
-
-```rust,ignore
-repository::create_wasm_repository!(CipherRepository, Cipher, "Repository<Cipher>");
-
-#[wasm_bindgen]
-impl StateClient {
-    pub fn register_cipher_repository(&self, store: CipherRepository) {
-        let store = store.into_channel_impl();
-        self.0.platform().state().register_client_managed(store)
-    }
+```rust
+macro_rules! create_client_managed_repositories {
+    ($container_name:ident, $macro:ident) => {
+        $macro! {
+            $container_name;
+            // List any SDK-managed repositories here. The format is:
+            // <fully qualified path to the item>, <item type idenfier>, <field name>, <name of the repository implementation>
+            ::bitwarden_vault::Cipher, Cipher, cipher, CipherRepository;
+            ::bitwarden_vault::Folder, Folder, folder, FolderRepository;
+        }
+    };
```

Comment:
praise: Nice idea! Using macros do define the repositories in a place where we're not actually able to depend on them 👍